tests/e2e/utils/kubeflow_installation.py

```diff
@@ -186,6 +188,46 @@ def build_alb_controller(cluster_name):
             --version v1.4.3".split()
     return subprocess.call(cmd)
 
+def exec_shell(cmd):
+    retcode = subprocess.call(cmd.split())
+    if retcode != 0:
+        raise Exception(f"ERROR: Failed to execute shell command \n{cmd}")
+
+def get_variable_from_params(path, var_name):
+    with open(path) as f:
+        for line in f:
+            if var_name in line:
+                return line.split("=")[1].strip()
+
+def find_and_replace_in_file(path, old_val, new_val):
+    with open(path, 'r') as file :
+        filedata = file.read()
+    filedata = filedata.replace(old_val, new_val)
+    with open(path, 'w') as file:
+        file.write(filedata)
+
+def build_ack_controller():
+    SERVICE="sagemaker"
+    RELEASE_VERSION="v0.4.3"
+    CHART_EXPORT_PATH="/tmp/chart"
+    CHART_REF=f"{SERVICE}-chart"
+    CHART_REPO=f"public.ecr.aws/aws-controllers-k8s/{CHART_REF}"
+    CHART_PACKAGE=f"{CHART_REF}-{RELEASE_VERSION}.tgz"
+    ACK_K8S_NAMESPACE="ack-system"
+    PARAMS_PATH="../../awsconfigs/common/ack-sagemaker-controller/params.env"
+    IAM_ROLE_ARN_FOR_IRSA=get_variable_from_params(PARAMS_PATH, "ACK_SAGEMAKER_OIDC_ROLE")
+    ACK_AWS_REGION=get_variable_from_params(PARAMS_PATH, "ACK_AWS_REGION")
+    LABEL_STRING='rbac.authorization.kubeflow.org/aggregate-to-kubeflow-edit: "true"'
+    
+    exec_shell(f"helm pull oci://{CHART_REPO} --version {RELEASE_VERSION} -d {CHART_EXPORT_PATH}")
```

Comment:
improvement for later: instead of cloning the repo locally, we should just install the controller and then set the IAM permissions similar to

https://aws-controllers-k8s.github.io/community/docs/user-docs/irsa/#step-3-associate-an-iam-role-to-a-service-account-and-restart-deployment

Reply:
ya
`helm install` requires logging into ecr public btw